src/Hyperion/Extensions/TypeEx.cs

```diff
@@ -118,11 +118,23 @@ private static Type GetTypeFromManifestName(Stream stream, DeserializerSession s
             return TypeNameLookup.GetOrAdd(byteArr, b =>
             {
                 var shortName = StringEx.FromUtf8Bytes(b.Bytes, 0, b.Bytes.Length);
+#if NET45
```

Comment:
And this is where all of the heavy-lifting for working between frameworks occurs I take it...